RadarSDK/Radar.h

```diff
@@ -121,56 +122,63 @@ typedef NS_ENUM(NSInteger, RadarRouteUnits) {
  
  @see https://radar.io/documentation/sdk#ios-foreground
  */
-typedef void(^ _Nullable RadarLocationCompletionHandler)(RadarStatus status, CLLocation * _Nullable location, BOOL stopped);
+typedef void (^_Nullable RadarLocationCompletionHandler)(RadarStatus status, CLLocation *_Nullable location, BOOL stopped);
 
 /**
  Called when a track request succeeds, fails, or times out. Receives the request status and, if successful, the user's location, an array of the events generated, and the user.
  
  @see https://radar.io/documentation/sdk#ios-foreground
  */
-typedef void(^ _Nullable RadarTrackCompletionHandler)(RadarStatus status, CLLocation * _Nullable location, NSArray<RadarEvent *> * _Nullable events, RadarUser * _Nullable user);
+typedef void (^_Nullable RadarTrackCompletionHandler)(RadarStatus status, CLLocation *_Nullable location, NSArray<RadarEvent *> *_Nullable events, RadarUser *_Nullable user);
 
 /**
  Called when a context request succeeds, fails, or times out. Receives the request status and, if successful, the location and the context.
  
  @see https://radar.io/documentation/api#context
  */
-typedef void(^ _Nonnull RadarContextCompletionHandler)(RadarStatus status, CLLocation * _Nullable location, RadarContext * _Nullable context);
+typedef void (^_Nonnull RadarContextCompletionHandler)(RadarStatus status, CLLocation *_Nullable location, RadarContext *_Nullable context);
 
 /**
  Called when a place search request succeeds, fails, or times out. Receives the request status and, if successful, the location and an array of places sorted by distance.
  
  @see https://radar.io/documentation/api#search-places
  */
-typedef void(^ _Nonnull RadarSearchPlacesCompletionHandler)(RadarStatus status, CLLocation * _Nullable location, NSArray<RadarPlace *> * _Nullable places);
+typedef void (^_Nonnull RadarSearchPlacesCompletionHandler)(RadarStatus status, CLLocation *_Nullable location, NSArray<RadarPlace *> *_Nullable places);
 
 /**
  Called when a geofence search request succeeds, fails, or times out. Receives the request status and, if successful, the location and an array of geofences sorted by distance.
 
  @see https://radar.io/documentation/api#search-geofences
  */
-typedef void(^ _Nonnull RadarSearchGeofencesCompletionHandler)(RadarStatus status, CLLocation * _Nullable location, NSArray<RadarGeofence *> * _Nullable geofences);
+typedef void (^_Nonnull RadarSearchGeofencesCompletionHandler)(RadarStatus status, CLLocation *_Nullable location, NSArray<RadarGeofence *> *_Nullable geofences);
```

Comment:
thinking through this a bit more, do you think it's possible to have a generic `RadarSearchCompletionHandler` that takes in an underlying entity class on which the operation searches? I can see that as a clean way to scale our search offering as we expand this out

Reply:
due to the restriction of ObjC, no generic is allowed for the typedef.... 

So we could either go verbose like this, or use a generic `NSArray *` for the callback. 

A slightly hacky way to avoid these boilerplate code is using macro

```
#define RADAR_SEARCH_COMPLETION_HANDLER(_contextType) ....

RADAR_SEARCH_COMPLETION_HANDLER(RadarGeofence);
RADAR_SEARCH_COMPLETION_HANDLER(RadarPoint)
```